File: backend/app/schemas/forecast.py
"""Pydantic schemas for Demand Forecast and Planner Review endpoints."""
from typing import List, Optional
from pydantic import BaseModel, Field


class ForecastSKUListItem(BaseModel):
    sku_id: str
    sku_name: str
    category: str
    forecast_run_id: str
    wape: float
    bias: float
    model_name: str
    review_status: Optional[str] = None


class SalesObservation(BaseModel):
    sales_date: str
    quantity_sold: int


class ForecastObservation(BaseModel):
    forecast_date: str
    forecast_qty: int


class ReviewItem(BaseModel):
    review_id: str
    forecast_run_id: str
    reviewer_username: str
    review_status: str
    planner_comment: str
    reviewed_at: str


class ForecastSKUDetailResponse(BaseModel):
    sku_id: str
    sku_name: str
    category: str
    analysis_date: str
    actual_start_date: str
    actual_end_date: str
    actual_sales: List[SalesObservation]
    forecast_start_date: str
    forecast_end_date: str
    forecast_results: List[ForecastObservation]
    forecast_run_id: str
    model_name: str
    wape: float
    bias: float
    evaluation_window_days: int
    latest_review: Optional[ReviewItem] = None
    review_history: List[ReviewItem] = Field(default_factory=list)


class CreateReviewRequest(BaseModel):
    review_status: str
    planner_comment: str

File: src/api/forecast.ts
import apiClient from './client';

export interface ForecastSKUListItem {
  sku_id: string;
  sku_name: string;
  category: string;
  forecast_run_id: string;
  wape: number;
  bias: number;
  model_name: string;
  review_status?: string | null;
}

export interface SalesObservation {
  sales_date: string;
  quantity_sold: number;
}

export interface ForecastObservation {
  forecast_date: string;
  forecast_qty: number;
}

export interface ReviewItem {
  review_id: string;
  forecast_run_id: string;
  reviewer_username: string;
  review_status: string;
  planner_comment: string;
  reviewed_at: string;
}

export interface ForecastSKUDetailResponse {
  sku_id: string;
  sku_name: string;
  category: string;
  analysis_date: string;
  actual_start_date: string;
  actual_end_date: string;
  actual_sales: SalesObservation[];
  forecast_start_date: string;
  forecast_end_date: string;
  forecast_results: ForecastObservation[];
  forecast_run_id: string;
  model_name: string;
  wape: number;
  bias: number;
  evaluation_window_days: number;
  latest_review?: ReviewItem | null;
  review_history: ReviewItem[];
}

export interface CreateReviewPayload {
  review_status: string;
  planner_comment: string;
}

export async function getForecastSKUs(): Promise<ForecastSKUListItem[]> {
  const res = await apiClient.get<ForecastSKUListItem[]>('/api/forecast/skus');
  return res.data;
}

export async function getForecastSKUDetail(skuId: string): Promise<ForecastSKUDetailResponse> {
  const res = await apiClient.get<ForecastSKUDetailResponse>(`/api/forecast/${skuId}`);
  return res.data;
}

export async function submitPlannerReview(skuId: string, payload: CreateReviewPayload): Promise<ReviewItem> {
  const res = await apiClient.post<ReviewItem>(`/api/forecast/${skuId}/review`, payload);
  return res.data;
}
